auth: kill auid

liewegas opened the pull request.

gregsfortytwo commented on qa/workunits/rados/test_rados_tool.sh:
```diff
@@ -153,10 +153,10 @@ VAL=`"$RADOS_TOOL" -p "$POOL" getxattr foo "rados.toothbrush"`
-run_expect_succ "$RADOS_TOOL" mkpool "$POOL_CP_TARGET"
+run "$CEPH_TOOL" osd pool rm "$POOL" "$POOL" --yes-i-really-really-mean-it
+run "$CEPH_TOOL" osd pool rm "$POOL_CP_TARGET" "$POOL_CP_TARGET" --yes-i-really-really-mean-it
+run_expect_succ "$CEPH_TOOL" osd pool crate "$POOL" 8
```
s/crate/create/

gregsfortytwo commented on src/tools/rados/rados.cc:
```diff
@@ -67,11 +67,6 @@ void usage(ostream& out)
 "   lspools                          list pools\n"
-"   mkpool <pool-name> [123[ 4]]     create pool <pool-name>'\n"
-"                                    [with auid 123[and using crush rule 4]]\n"
-"   cppool <pool-name> <dest-pool>   copy content of a pool\n"
```
I don't like the cppool command, but it does see some use in the community and at a quick skim of the mailing list I'm not sure all that use is broken. :/ I know you asked about it on the auid dev list thread, but we may need to discuss that more broadly.

gregsfortytwo commented on src/osd/OSDCap.cc:
```diff
@@ -85,10 +85,6 @@ ostream& operator<<(ostream &out, const OSDCapPoolTag &pt)
-  if (m.auid != -1LL) {
-    out << "auid " << m.auid << " ";
-  }
-
```
The code changes all look fine, but this commit message includes a separate line "fix osdcap" after the s-o-b statement?

gregsfortytwo commented on doc/dev/cephx_protocol.rst:
```diff
@@ -146,15 +146,17 @@ give it an authentication ticket.
+ID to the global ID in that old ticket.  If the attempt to decode its
+old ticket fails (most probably because it didn't have one),
+``should_enc_ticket`` remains false.  Now we set up the new ticket,
+filling in timestamps, the name of C, the global ID provided in the
```
Should be ", *and* the global ID"

liewegas commented on src/librados/librados.cc:
```diff
@@ -2423,13 +2426,19 @@ int librados::Rados::pool_create(const char *name)
+  return client->pool_create(str, crush_rule);
+}
+
+int librados::Rados::pool_create_with_rule(const char *name, __u8 crush_rule)
```
yeah, everything that takes an auid is marked deprecated in the previous commit

liewegas commented on src/librados/librados.cc:
```diff
@@ -2423,13 +2426,19 @@ int librados::Rados::pool_create(const char *name)
 int librados::Rados::pool_create(const char *name, uint64_t auid)
 {
   string str(name);
-  return client->pool_create(str, auid);
```
yeah good idea.  i went with EINVAL though since it's more like 'invalid input'.. it seems weird to have a EOPNOTSUPP return value that depends on the arguments passed.

liewegas commented on src/mon/OSDMonitor.cc:
```diff
@@ -5947,16 +5947,10 @@ int OSDMonitor::prepare_new_pool(MonOpRequestRef op)
   stringstream ss;
   string rule_name;
   int ret = 0;
-  if (m->auid)
```
could, but i removed the auid field from the message too.  not sure it's worth resurrecting for this

liewegas commented:
Ok, i think I addressed all of the comments.  cppool is there there... I have a patch that removes it but i'll include it in a separate follow-on PR with a separate discussion.  Thanks for the detailed review!

- the moncap instances of auid are the "does not parse" test cases
- all librados auid calls are now marked deprecated
- i left the pg_pool_t member there for now.  i'm worried about subtle issues iwth pg_pool_t encoding discrepancies during upgrade.  there are no users so we can clean it out completely after a couple releases.

gregsfortytwo commented:
Sounds good! We can rename the auid field in pg_pool_t later.

tchaikov commented:
@liewegas needs rebase
